test/unit/buffer_spec.lua

```diff
@@ -293,11 +298,131 @@ describe('buffer functions', function()
     it('should print the number of lines in the buffer', function()
       buffer.setfname(globals.curbuf, to_cstr("test/unit/buffer_spec.lua"), NULL, 1)
 
-      local width = build_stl_str_hl("%L")
+      local width = build_stl_str_hl("%L", 80)
 
       eq(1, width)
       eq("1", helpers.ffi.string(output_buffer, width))
 
     end)
+
+    it('should right align when using =', function()
+      buffer.setfname(globals.curbuf, to_cstr("test/unit/buffer_spec.lua"), NULL, 1)
+
+      local expected_width = 20
+      local width = build_stl_str_hl("neo%=vim", expected_width)
+
+      eq(width, expected_width)
+      eq("neo              vim", get_str(output_buffer, expected_width))
+    end)
+
+    it('should approximately center text when using %=text%=', function()
+      buffer.setfname(globals.curbuf, to_cstr("test/unit/buffer_spec.lua"), NULL, 1)
+
+      local expected_width = 20
+      local width = build_stl_str_hl("abc%=neovim%=def", expected_width)
+
+      eq(width, expected_width)
+      eq("abc    neovim    def", get_str(output_buffer, expected_width))
+    end)
```

Comment:
Can you also repeat with `expected_width = 21` (i.e. so that spaces cannot be possibly distributed equally)? I also still do not see tests like `foobar%=test%=baz` (non-equal width on sides), `foobar%=test%=`/`%=test%=baz` (empty side), `%=test%=` (empty sides).